core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -327,7 +336,12 @@ public PartitionSpec apply() {
 
   @Override
   public void commit() {
-    TableMetadata update = base.updatePartitionSpec(apply());
+    TableMetadata update;
+    if (this.setAsDefault) {
```

Comment:
typically in the project, `this` is only used when writing to the field. Here we could just reference `setAsDefault`